docs/modules/ROOT/pages/api/erc1155.adoc

```diff
@@ -0,0 +1,664 @@
+substitution mechanism defined in the ERC1155 standard.
+
+See https://eips.ethereum.org/EIPS/eip-1155#metadata.
+
+By this mechanism, any occurrence of the `\{id\}` substring in either the
+URI or any of the values in the JSON file at said URI will be replaced by
+clients with the token type ID.
+
+For example, the pass:[<code>https://token-cdn-domain/\{id\}.json</code>] URI would be
+interpreted by clients as
+pass:[<code>https://token-cdn-domain/000000000000...000000000000004cce0.json</code>]
+for token type ID 0x4cce0.
+
+Because these URIs cannot be meaningfully represented by the `URI` event,
+this function emits no events.
+
+==== Events
+
+[.contract-item]
+[[ERC1155Component-TransferSingle]]
+==== `[.contract-item-name]#++TransferSingle++#++(operator: ContractAddress, from: ContractAddress, to: ContractAddress, id: u256, value: u256)++` [.item-kind]#event#
+
+See <<IERC1155-TransferSingle,IERC1155::TransferSingle>>.
+
+[.contract-item]
+[[ERC1155Component-TransferBatch]]
+==== `[.contract-item-name]#++TransferBatch++#++(operator: ContractAddress, from: ContractAddress, to: ContractAddress, ids: Span<u256>, values: Span<u256>)++` [.item-kind]#event#
+
+See <<IERC1155-TransferBatch,IERC1155::TransferBatch>>.
+
+[.contract-item]
+[[ERC1155Component-ApprovalForAll]]
+==== `[.contract-item-name]#++ApprovalForAll++#++(owner: ContractAddress, operator: ContractAddress, approved: bool)++` [.item-kind]#event#
+
+See <<IERC1155-ApprovalForAll,IERC1155::ApprovalForAll>>.
+
+[.contract-item]
+[[ERC1155Component-URI]]
+==== `[.contract-item-name]#++URI++#++(value: ByteArray, id: u256)++` [.item-kind]#event#
+
+See <<IERC1155-URI,IERC1155::URI>>.
+
+== Receiver
+
+[.contract]
+[[IERC1155Receiver]]
+=== `++IERC1155Receiver++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.9.0/src/token/erc1155/interface.cairo[{github-icon},role=heading-link]
+
+[.hljs-theme-dark]
+```javascript
+use openzeppelin::token::erc1155::interface::IERC1155Receiver;
+```
+
+Interface for contracts that support receiving token transfers from `ERC1155` contracts.
+
+[.contract-index]
+.{inner-src5}
+--
+0x15e8665b5af20040c3af1670509df02eb916375cdf7d8cbaf7bd553a257515e
+--
+
+[.contract-index]
+.Functions
+--
+* xref:#IERC1155Receiver-on_erc1155_received[`++on_erc1155_received(operator, from, token_id, value, data)++`]
+* xref:#IERC1155Receiver-on_erc1155_batch_received[`++on_erc1155_batch_received(operator, from, token_ids, values, data)++`]
+--
+
+==== Functions
+
+[.contract-item]
+[[IERC1155Receiver-on_erc1155_received]]
+==== `[.contract-item-name]#++on_erc1155_received++#++(operator: ContractAddress, from: ContractAddress, token_id: u256, value: u256, data Span<felt252>) -> felt252++` [.item-kind]#external#
+
+Whenever an IERC1155 `token_id` token is transferred to this non-account contract via <<IERC1155-safe_transfer_from,IERC1155::safe_transfer_from>> by `operator` from `from`, this function is called.
```

Comment:
this contract is not necessarily a non-account. if an account supports the receiver interface, this method will be called on it